src/uct/api/uct_def.h

```diff
@@ -111,6 +117,13 @@ typedef int                      uct_worker_cb_id_t;
     |<---- stride ----->|<---- stride ----->|
    @endverbatim
  *
+ * In order to describe an interleaved memory layout, each structure contains
```

Comment:
Correction: where do you state you have just 2 types, A & B, and not A & B & C ? I see you're using count for the length, rather than the number of dtype descriptions.